singularity/configure.py

```diff
@@ -51,25 +53,35 @@ def export_file(file_path, content):
 
     return
 
-def configure_rabbit(rabbit_port, ca_path, cert_path, key_path, config_path):
-    configuration = '''[
-  {ssl, [{versions, ['tlsv1.2', 'tlsv1.1']}]},
-  {kernel, [
-    {inet_dist_listen_min, 33672},
-    {inet_dist_listen_max, 33672}
-  ]},
-  {rabbit2, [
-    {ssl_listeners, [%(rabbit_port)s]},
-    {ssl_options, [{cacertfile, "%(ca_path)s"},
-                   {certfile, "%(cert_path)s"},
-                   {keyfile, "%(key_path)s"},
-                   {versions, ['tlsv1.2','tlsv1.1']},
-                   {verify, verify_peer},
-                   {fail_if_no_peer_cert, true}]}
-   ]}
-].''' % { 'rabbit_port' : rabbit_port, 'ca_path' : ca_path, 'cert_path' : cert_path, 'key_path' : key_path }
-
-    return export_file(os.path.join(config_path,'rabbitmq.config'), configuration)
+def configure_rabbit(rabbit_port, rabbit_name, rabbit_dist, ca_path, cert_path, key_path, config_path, dpdash_path):
+    rabbit_path = os.path.join(dpdash_path, 'rabbitmq')
+    rabbit_conf_path = os.path.join(config_path,'rabbitmq.conf')
+    rabbit_env_path = os.path.join(config_path,'rabbitmq-env.conf')
+
+    # Generate and export rabbitmq-env.conf
+    env = '''NODENAME=%(rabbit_name)s
+NODE_PORT=%(rabbit_port)s
+DIST_PORT=%(rabbit_dist)s
+CTL_DIST_PORT_MIN=%(rabbit_dist_min)s
+CTL_DIST_PORT_MAX=%(rabbit_dist_max)s
+MNESIA_BASE=%(rabbit_path)s
+LOG_BASE=%(rabbit_path)s
+''' % { 'rabbit_name' : rabbit_name, 'rabbit_port' : rabbit_port, 'rabbit_dist': rabbit_dist, 'rabbit_dist_min': rabbit_dist+10000, 'rabbit_dist_max': rabbit_dist+10010, 'rabbit_path': rabbit_path }
+
+    export_file(rabbit_env_path, env)
+
+    # Generate and export rabbitmq.conf
+    configuration = '''ssl_options.versions.1           = tlsv1.2
+ssl_options.versions.2           = tlsv1.1
+ssl_options.cacertfile           = %(ca_path)s
+ssl_options.certfile             = %(cert_path)s
+ssl_options.keyfile              = %(key_path)s
+ssl_options.verify               = verify_peer
+ssl_options.fail_if_no_peer_cert = true
+listeners.ssl.default            = %(rabbit_port)s
```

Comment:
As shown [here](https://www.rabbitmq.com/networking.html#epmd-inet-dist-port-range):

```
  {kernel, [
    {inet_dist_listen_min, 33672},
    {inet_dist_listen_max, 33672}
  ]}
```

Can we add:

```cfg
kernel.inet_dist_listen_min    =%(rabbit_dist_min)s
kernel.inet_dist_listen_max    =%(rabbit_dist_max)s
```

I am not saying we should because I am not confident if the new style would be respected.

Reply:
As you posted above, the docs claim those can only be set in the old style. So I don't think it would read that from the config file in the new style—we would have to have a separate file ending in `.config` that uses the old style. However, the same doc ([Inter-node Communication Port Range](https://www.rabbitmq.com/networking.html#epmd-inet-dist-port-range)) also says:

> This range can be configured using the RABBITMQ_CTL_DIST_PORT_MIN and RABBITMQ_CTL_DIST_PORT_MAX environment variables. 

As I said in Teams, this won't actually do anything until we upgrade to 3.8.x. I thought the default behavior would be fine until then, though, since these weren't set manually before your commit the other day.

Reply:
I had to use `DIST_PORT` to let you run another instance. Otherwise, it was clashing with mine.

Reply:
Ah, I see. I'll try it with setting those in the new config style as you've posted, and if it doesn't work, then just use an additional `.config` file in the old style.

Reply:
Actually, that would require you to set another env var for `*.config` to be found, won't it?

Reply:
Correct, it would be `advanced.config` and env variable `RABBITMQ_ADVANCED_CONFIG_FILE`.

Reply:
If that is the case, let's not spend time there. We can just depend on `DIST_PORT` and whatever mystic calculation `rabbitmq` does to obtain the other two from `DIST_PORT`.

Reply:
OOPS—actually, it looks like `advanced.config` wasn't added until 3.7.x. So we would have to pick one config style or the other for the primary config file.

I'll see if it picks it up from just using `kernel.inet_dist_listen_min` and `max` in the new style. Though, if it doesn't work, it might not matter—I was able to run it just fine without collision, and I had let it use the default behavior.